### PR TITLE
Remove grant select from cdb tablemetadata to public

### PR DESCRIPTION
**Pull request by rochoa**
- Select permission in CDB_TableMetadata no longer granted to public
 - New function to upsert the updated_at in CDB_TableMetadata for a regclass

### CODE REVIEW

**Comment by rochoa** [comment]
cc @Kartones @javisantana review this one

**Comment by Kartones** [comment]
+1 (and sorry for the late CR)

File: scripts-available/CDB_TableMetadata.sql [hunk 1]
```diff
@@ -118,3 +118,21 @@ CREATE TRIGGER table_modified AFTER INSERT OR UPDATE
 ON CDB_TableMetadata FOR EACH ROW EXECUTE PROCEDURE
  _CDB_TableMetadata_Updated();
 
+
+-- similar to TOUCH(1) in unix filesystems but for table in cdb_tablemetadata
+CREATE OR REPLACE FUNCTION public.CDB_TableMetadataTouch(tablename regclass)
```
Comment (javisantana) [inline scripts-available/CDB_TableMetadata.sql]: I understand this function will only work for roles with permissions on cdb_tablemetadata, right?

**Comment by javisantana** [comment]
:es: for me, I left a question for clarification

File: scripts-available/CDB_TableMetadata.sql (hunk 1 above)
Comment (rochoa) [inline scripts-available/CDB_TableMetadata.sql]: Correct! Anyway I'm gonna add a test to cover that behaviour for future/potential regressions.

**Pull request merged**